api/handler/v1/runtime_test.go

```diff
@@ -3034,4 +3053,752 @@ func TestRuntimeServiceServer(t *testing.T) {
+			Name: projectName,
+		}
+		namespaceSpec := models.NamespaceSpec{
+			ID:   uuid.Must(uuid.NewRandom()),
+			Name: "dev-test-namespace-1",
+			Config: map[string]string{
+				"bucket": "gs://some_folder",
+			},
+			ProjectSpec: projectSpec,
+		}
+		resourceName := "a-data-project:dataset.table"
+		resourceUrn := "datastore://a-data-project:dataset.table"
+		backupUrn := "datastore://a-data-project:optimus_backup.table_backup"
+
+		t.Run("should able to do backup ignoring downstream", func(t *testing.T) {
+			projectRepository := new(mock.ProjectRepository)
+			defer projectRepository.AssertExpectations(t)
+
+			projectRepoFactory := new(mock.ProjectRepoFactory)
+			defer projectRepoFactory.AssertExpectations(t)
+
+			namespaceRepository := new(mock.NamespaceRepository)
+			defer namespaceRepository.AssertExpectations(t)
+
+			namespaceRepoFact := new(mock.NamespaceRepoFactory)
+			defer namespaceRepoFact.AssertExpectations(t)
+
+			resourceSvc := new(mock.DatastoreService)
+			defer resourceSvc.AssertExpectations(t)
+
+			jobService := new(mock.JobService)
+			defer jobService.AssertExpectations(t)
+
+			jobName := "a-data-job"
+			jobSpec := models.JobSpec{
+				ID:   uuid.Must(uuid.NewRandom()),
+				Name: jobName,
+				Task: models.JobSpecTask{
+					Config: models.JobSpecConfigs{
+						{
+							Name:  "do",
+							Value: "this",
+						},
+					},
+				},
+				Assets: *models.JobAssets{}.New(
+					[]models.JobSpecAsset{
+						{
+							Name:  "query.sql",
+							Value: "select * from 1",
+						},
+					}),
+			}
+			resourceSpec := models.ResourceSpec{
+				Name: resourceName,
+				URN:  resourceUrn,
+			}
+			backupRequestPb := pb.BackupRequest{
+				ProjectName:      projectName,
+				DatastoreName:    models.DestinationTypeBigquery.String(),
+				ResourceName:     resourceName,
+				Namespace:        namespaceSpec.Name,
+				IgnoreDownstream: true,
+				Config: map[string]string{
+					"TTL": "30",
+				},
+			}
+			backupReq := models.BackupRequest{
+				ResourceName: resourceName,
+				Project:      projectSpec,
+				Namespace:    namespaceSpec,
+				Config: map[string]string{
+					"TTL": "30",
+				},
+				DryRun:           false,
+				IgnoreDownstream: true,
+			}
+			backupResponsePb := &pb.BackupResponse{
+				Urn: []string{backupUrn},
+			}
+
+			projectRepository.On("GetByName", projectName).Return(projectSpec, nil)
+			projectRepoFactory.On("New").Return(projectRepository)
+
+			namespaceRepository.On("GetByName", namespaceSpec.Name).Return(namespaceSpec, nil)
+			namespaceRepoFact.On("New", projectSpec).Return(namespaceRepository)
+
+			resourceSvc.On("ReadResource", context.Background(), namespaceSpec, models.DestinationTypeBigquery.String(), resourceName).Return(resourceSpec, nil)
+			jobService.On("GetByDestination", projectSpec, resourceUrn).Return(jobSpec, nil)
+			resourceSvc.On("BackupResource", context.Background(), backupReq, []models.JobSpec{jobSpec}).Return([]string{backupUrn}, nil)
```

Comment:
because i excluded the `backupTime` in the mock 😄 but I think it is better to remove backupTime initialization from runtime and just do it in bigquery package, so will modify it.

Reply:
removed backupTime init from runtime